src/Tools/GenerateGlobalAnalyzerConfigs/Program.cs

```diff
@@ -347,22 +348,67 @@ static string GetGlobalAnalyzerConfigTargetContents(string packageName)
 ";
             }
 
-            static string GetMSBuildPropertyOptionItemGroup()
+            static string GetMSBuildContentForPropertyAndItemOptions()
             {
                 var builder = new StringBuilder();
 
-                builder.Append($@"
-  <!-- MSBuild properties to thread to the analyzers as options --> 
-  <ItemGroup>
-");
-                foreach (var field in typeof(MSBuildPropertyOptionNames).GetFields())
+                AddMSBuildContentForPropertyOptions(builder);
+                AddMSBuildContentForItemOptions(builder);
+
+                return builder.ToString();
+
+                static void AddMSBuildContentForPropertyOptions(StringBuilder builder)
                 {
-                    builder.AppendLine($@"    <CompilerVisibleProperty Include=""{field.Name}"" />");
+                    var compilerVisibleProperties = new List<string>();
+                    foreach (var field in typeof(MSBuildPropertyOptionNames).GetFields())
+                    {
+                        compilerVisibleProperties.Add(field.Name);
+                    }
+
+                    // Add ItemGroup for MSBuild property names that are required to be threaded as analyzer config options.
+                    AddItemGroupForCompilerVisibleProperties(compilerVisibleProperties, builder);
                 }
 
-                builder.AppendLine($@"  </ItemGroup>");
+                static void AddItemGroupForCompilerVisibleProperties(List<string> compilerVisibleProperties, StringBuilder builder)
+                {
+                    builder.AppendLine($@"
+  <!-- MSBuild properties to thread to the analyzers as options --> 
+  <ItemGroup>");
+                    foreach (var compilerVisibleProperty in compilerVisibleProperties)
+                    {
+                        builder.AppendLine($@"    <CompilerVisibleProperty Include=""{compilerVisibleProperty}"" />");
+                    }
 
-                return builder.ToString();
+                    builder.AppendLine($@"  </ItemGroup>");
+                }
+
+                static void AddMSBuildContentForItemOptions(StringBuilder builder)
+                {
+                    // Add ItemGroup and PropertyGroup for MSBuild item names that are required to be threaded as analyzer config options.
+                    // The analyzer config option will have the following key/value:
+                    // - Key: Item name prefixed with an '_' and suffixed with a 'List' to reduce chances of conflicts with any existing project property.
+                    // - Value: Concatenated item metadata values, separate by a ';' character. See https://github.com/dotnet/sdk/issues/12706#issuecomment-668219422 for details.
```

Comment:
NIT: didn't you mentioned the separator will be ',' instead of `;` in your [comment](https://github.com/dotnet/sdk/issues/12706#issuecomment-668768332)